cli/format/sort.go

```diff
@@ -52,3 +53,11 @@ func (s ByAddress) Less(i, j int) bool {
 	}
 	return bytes.Compare(net.ParseIP(ip1), net.ParseIP(ip2)) < 0
 }
+
+type SMatcher []*silencepb.Matcher
+
+func (m SMatcher) Len() int      { return len(m) }
+func (m SMatcher) Swap(i, j int) { m[i], m[j] = m[j], m[i] }
+func (m SMatcher) Less(i, j int) bool {
+	return m[i].Name < m[j].Name
```

Comment:
You need to make this sort stable as Alertmanager does not force name to be unique in silences. For example, `foo="bar"` and `foo=~[a-zA-Z]+` can be used in the same silence. I don't think a lot of users do this, but the sort is non-deterministic when two names are the same.

Reply:
For example `foo="bar"`, in order to stabilize the sort, we need to construct a string such as` "foo=bar"`, and then sort the result, not just `"foo"`